### PR TITLE
Non-functional code style changes.

### PR DESCRIPTION
**Pull request by benjamin-work**
Mostly to satisfy pylint. Also some minor refactorings to remove code
duplication (none of which affect public API) and addition of some
docstrings.

### CODE REVIEW

File: skorch/callbacks/training.py [hunk 1]
```diff
@@ -223,9 +223,11 @@ def f_history_(self):
...
         """Returns a dictionary of formatted filenames"""
         idx = -1
-        if (self.event_name is not None and
-           net.history is not None and
-           len(net.history) > 0):
+        if (
+                self.event_name is not None and
+                net.history is not None and
+                net.history
+        ):
```
Comment (thomasjpfan) [inline skorch/callbacks/training.py]: Could this be:

```python
if self.event_name is not None and net.history:
```

File: skorch/tests/test_utils.py [hunk 2]
```diff
@@ -46,7 +46,14 @@ def tensors_equal(self, x, y):
 
         return (x == y).all()
 
+    # pylint: disable=no-method-argument
     def parameters():
+        """Yeilds data, expected value, and device for tensor conversion
```
Comment (thomasjpfan) [inline skorch/tests/test_utils.py]:
```suggestion
        """Yields data, expected value, and device for tensor conversion
```

**Comment by benjamin-work** [comment]
@ottonemo pls review again

File: skorch/callbacks/lr_scheduler.py [hunk 3]
```diff
@@ -230,7 +235,7 @@ def get_lr(self):
         return current_lrs.tolist()
 
 
-class CyclicLR(object):
+class CyclicLR:
```
[the remark this answers is not in the document]
Comment (benjamin-work) [reply skorch/callbacks/lr_scheduler.py]: In Python 3, there are no old-style classes anymore.

[the remark this answers is not in the document]
Comment (taketwo) [reply skorch/callbacks/lr_scheduler.py]: Thanks, I was not aware of this. Sorry for the noise!

For the record, there are two more classes explicitly deriving from `object` in "examples/word_language_model/data.py".

**Comment by benjamin-work** [comment]
@ottonemo pls review again

**Pull request merged**